acceptance/tests/plan_ssh.rb

```diff
@@ -7,58 +8,95 @@
+  user = ENV['SSH_USER']
+  password = ENV['SSH_PASSWORD']
+  nodes_csv = ssh_nodes.map(&:hostname).join(',')
+
   dir = bolt.tmpdir('C100553')
 
-  step "create plan on bolt controller" do
+  step 'create task on bolt controller' do
     on(bolt, "mkdir -p #{dir}/modules/test/{tasks,plans}")
-    create_remote_file(bolt, "#{dir}/modules/test/tasks/a_unix", <<-FILE)
-    echo "Line one from task a" > /tmp/C100553_plan_artifact.txt
-    FILE
-    create_remote_file(bolt, "#{dir}/modules/test/tasks/b_unix", <<-FILE)
-    echo "Line two from task b" >> /tmp/C100553_plan_artifact.txt
+    create_remote_file(bolt, "#{dir}/modules/test/tasks/hostname_nix.sh", <<-FILE)
+#!/bin/bash
+if [ ! -f /tmp/retry.txt ] && [ "$HOSTNAME" = '#{node}' ]; then
+  touch /tmp/retry.txt;
+  exit 1
+elif [ -f /tmp/retry.txt ]; then
+  rm /tmp/retry.txt
+fi
     FILE
+  end
+
+  step "create plan on bolt controller" do
     create_remote_file(bolt,
                        "#{dir}/modules/test/plans/my_unix_plan.pp", <<-FILE)
-    plan test::my_unix_plan($nodes) {
-      $nodes_array = $nodes.split(',')
-      notice("${run_task(test::a_unix, $nodes_array)}")
-      notice("${run_task(test::b_unix, $nodes_array)}")
-    }
+plan test::my_unix_plan($nodes) {
+  $array = $nodes.split(',')
+  $result = run_task(test::hostname_nix, $array,
+        'success_nodes' => [$array[0]],
+        '_catch_errors' =>true)
+  $retry = run_task(test::hostname_nix, $result.error_set.names,
+        'success_nodes' =>  $result.error_set.names)
+  return({ 'result' => $result, 'retry' => $retry })
+}
     FILE
   end
 
-  step "execute `bolt plan run` via SSH" do
-    user = ENV['SSH_USER']
-    password = ENV['SSH_PASSWORD']
-    nodes_csv = ssh_nodes.map(&:hostname).join(',')
+  step "execute `bolt plan run` via SSH with json output" do
     bolt_command = "bolt plan run test::my_unix_plan nodes=#{nodes_csv}"
-
     flags = {
       '-u'                     => user,
       '--modulepath'           => "#{dir}/modules",
       '-p'                     => password,
+      '--format'               => 'json',
       '--no-host-key-check'    => nil
     }
 
     result = bolt_command_on(bolt, bolt_command, flags)
-    message = "The plan was expected to notify but did not"
-    assert_match(/Notice:/, result.stdout, message)
-    ssh_nodes.each do |node|
-      message = "The plan was expected to mention the host #{node.hostname} with _output"
-      assert_match(/"node":"#{node.hostname}","status":"success"/, result.stdout, message)
-    end
+    assert_equal(0, result.exit_code,
+                 "The task did not exit with code 0 as expected")
 
-    ssh_nodes.each do |node|
-      # bolt plan return value unspecified
-      command = "cat /tmp/C100553_plan_artifact.txt"
-      on(node, command, accept_all_exit_codes: true) do |res|
-        cat_fail_msg = "The 'cat' command was not successful"
-        assert_equal(res.exit_code, 0, cat_fail_msg)
+    json = JSON.parse(result.stdout)
```

Comment:
Try and making a failing asserting in the rescue has been working better for json parsing.